### ISSUE TITLE
Add different status codes for plan

### ISSUE THREAD

**Issue by marcosnils** [mantika/whaleprint#9]
If there's nothing to do we could return 0 meaning "everything is ok" and otherwise, some other code that means "there are changes pending in your services"

Right now we're returning 0 for both. 

@xetorthio WDYT?

**Comment by xetorthio** [comment]
Not sure. What would be the use case?

**Comment by marcosnils** [comment]
As part of your CI deployment you make a bundle out of the compose-file and you can realize if you have some service changes by getting the status code of whaleprint plan for example.

**Comment by xetorthio** [comment]
Ok. Sounds interesting. Do we know what terraform does?

**Comment by marcosnils** [comment]
terraform returns 0 always even if there are changes.

**Comment by leokhoa** [comment]
IMHO, I think it's useful if we have different status codes.

**Comment by rayjohnson** [comment]
I think it is more standard for non-zero exit codes to mean something is wrong (like dab file corrupted and can not be parsed.)  Rather than "I have changes to make vs. no changes".  

But certainly having output that is easy to parse for the tool to be used in CI or other automation would be very useful!

**Comment by xetorthio** [comment]
Yeah. I agree. Exit code should imply that something went wrong and "having differences" doesn't mean it went wrong.
Now, I am still not sure about the use case, generating a new DAB will always show changes in whaleprint, because a new DAB probably means a new image, right? So there will be always changes.

**Comment by marcosnils** [comment]
No necessary. What if you make a configuration change that doesn't involve building a new image?. 

Is there a way we can return something to the user so he can programmatically check if something changed?

**Comment by xetorthio** [comment]
Can you provide a more specific example? What do you mean by configuration change?
I guess what you want to do is to deploy automatically as long as there are no changes? That's what you want to achieve with your CI?
